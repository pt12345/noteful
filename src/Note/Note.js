import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'
import { withRouter } from 'react-router-dom'
import { format, parseISO } from 'date-fns'
import PropTypes from 'prop-types'

class Note extends Component {

  static defaultProps ={
    onDeleteNote: () => {},
  }

  static contextType = ApiContext;

  handleDelete = event => {
    event.preventDefault();
    const noteId = this.props.id

    fetch(`http://localhost:9090/notes/${noteId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    })
    .then(response => {
      if(!response.ok)
        return response.json().then(event => Promise.reject(event))
      
      return response.json()
    })
    .then(responseJson => {
      
      this.context.deleteNote(noteId)
      this.props.onDeleteNote(noteId)
    })
    .catch(error => {
      console.log({error})
    })  
  }

  render() {

    let modified = ""
    try {
      modified = format(parseISO(this.props.modified), 'MMM yyyy')
    }
    catch {
      modified = "N/A"
    }

    return(
      <div className="Note">
          <h2>
            <Link to={`/note/${this.props.id}`}>
                {this.props.name}
            </Link>
          </h2>
          <div className="note_container">
            <div className="date_modified">
            Last Modified: {modified}
            </div>
            <div className='delete_button'>
              <button type='button' onClick={this.handleDelete}>
                Delete Note
              </button>
            </div>
          </div>  
      </div>
    );
  }
}

Note.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  modified: PropTypes.string
};

export default withRouter(Note);