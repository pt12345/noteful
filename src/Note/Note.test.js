import React from 'react';
import ReactDOM from 'react-dom';
import Note from './Note';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

describe('Note', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<BrowserRouter><Note /></BrowserRouter>, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const note = renderer.create(<BrowserRouter><Note /></BrowserRouter>)
    expect(note).toMatchSnapshot();
  })

})
