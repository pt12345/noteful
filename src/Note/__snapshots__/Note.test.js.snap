// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders the UI as expected 1`] = `
<div
  className="Note"
>
  <h2>
    <a
      href="/note/undefined"
      onClick={[Function]}
    />
  </h2>
  <div
    className="note_container"
  >
    <div
      className="date_modified"
    >
      Last Modified: 
      N/A
    </div>
    <div
      className="delete_button"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Delete Note
      </button>
    </div>
  </div>
</div>
`;
